### PR TITLE
MGMT-18965: Update docker files to use universal base image

### PR DESCRIPTION
**Pull request by paul-maidment**
This updates the docker files of this project to use the universal base image

### CODE REVIEW

File: Dockerfile.assisted-installer [hunk 1]
```diff
@@ -1,16 +1,20 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.20 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.20 AS builder
 ARG TARGETPLATFORM
 ENV GOFLAGS=-mod=mod
 WORKDIR /go/src/github.com/openshift/assisted-installer
 
+# Workaround for creating build folder
```
Comment (rccrdpccl) [inline Dockerfile.assisted-installer:6]: Why using a workaround when we can use this image the way it's meant to?

**Comment by danmanor** [comment]
@paul-maidment  Can you explain the reasoning for this branch ?

**Comment by paul-maidment** [comment]
> @paul-maidment Can you explain the reasoning for this branch ?

The 4.16 branch is broken unless we do something, try running

`skipper make build` on it

Presumably moving to UBI would be the correct course of action?

File: Dockerfile.assisted-installer (hunk 1 above)
Comment (paul-maidment) [reply Dockerfile.assisted-installer:6]: It's not being used correctly on master?

https://github.com/openshift/assisted-installer/blob/master/Dockerfile.assisted-installer#L7

**Comment by paul-maidment** [comment]
No longer needed, it's possible to work around this for now

**Pull request closed without merging**